overlord/boot/firstboot.go

```diff
@@ -111,31 +94,14 @@ func populateStateFromSeed() error {
 		return nil
 	}
 
-	st.Lock()
-	msg := fmt.Sprintf("First boot seeding")
+	msg := fmt.Sprintf("Initialize system state")
 	chg := st.NewChange("seed", msg)
 	for _, ts := range tsAll {
 		chg.AddAll(ts)
 	}
-	st.Unlock()
-
-	// do it and wait for ready
-	ovld.Loop()
-
-	st.EnsureBefore(0)
-	<-chg.Ready()
+	snapstate.SetGlobalFlag(st, snapstate.Seeded)
```

Comment:
As discussed I reworked this to a single flag in the state. Much simpler and nicer this way.